pkg/kube/snapshot/snapshot.go

```diff
@@ -153,3 +157,25 @@ func NewSnapshotter(kubeCli kubernetes.Interface, dynCli dynamic.Interface) (Sna
 	}
 	return nil, errors.New("Snapshot resources not supported")
 }
+
+// We use regexp to match because errors written in vs.Status.Error.Message are strings
+// and we don't have any status code or other metadata in there.
+var transientErrorRegexp = regexp.MustCompile("the object has been modified; please apply your changes to the latest version and try again")
+
+// Use regexp to detect resource conflict error
+// If CSI snapshotter changes error reporting to use more structured errors,
+// we can improve this function to parse and recognise error codes or types.
+func isTransientError(err error) bool {
+	return transientErrorRegexp.MatchString(err.Error())
+}
+
+func waitOnReadyToUse(ctx context.Context, dynCli dynamic.Interface, snapGVR schema.GroupVersionResource, snapshotName, namespace string, isReadyFunc func(*unstructured.Unstructured) (bool, error)) error {
```

Comment:
```suggestion
func waitOnReadyToUse(
	ctx context.Context,
	dynCli dynamic.Interface,
	snapGVR schema.GroupVersionResource,
	snapshotName,
	namespace string,
	isReadyFunc func(*unstructured.Unstructured) (bool, error),
) error {
```
Trying to slowly incorporate [this](https://thanos.io/tip/contributing/coding-style-guide.md/#only-two-ways-of-formatting-functionsmethods) coding style for readability improvements